source/hp/fe_collection.cc

```diff
@@ -314,6 +314,50 @@ namespace hp
 
 
 
+  template <int dim, int spacedim>
+  std::deque<unsigned int>
+  FECollection<dim, spacedim>::get_hierarchy_sequence(
+    const unsigned int fe_index) const
+  {
+    AssertIndexRange(fe_index, size());
+
+    std::deque<unsigned int> sequence = {fe_index};
+
+    // get predecessors
+    {
+      unsigned int previous, front = sequence.front();
+      while ((previous = previous_in_hierarchy(front)) != front)
+        {
+          sequence.push_front(previous);
+          front = previous;
+
+          Assert(sequence.size() <= finite_elements.size(),
+                 ExcMessage(
+                   "The registered hierarchy is not terminated: "
+                   "previous_in_hierarchy() does not stop at a final index."));
```

Comment:
That would be a funny thing to happen ;-)

Reply:
Users could make mistakes with setting up their custom hierarchy correctly, that is why I left this exception here.

Reply:
I'm just getting very careful whenever I introduce `while` loops ;-)

Reply:
You know how I feel about assertions ;-)